cfme/tests/intelligence/reports/test_validate_chargeback_report.py

```diff
@@ -205,18 +207,32 @@ def query_rate(db, provider, metric, description, rate_type):
     details = db['chargeback_rate_details']
     rates = db['chargeback_rates']
 
+    def add_rate(tiered_rate):
+        list_of_rates.append(tiered_rate)
+
     with db.transaction:
-        providers = (
+        result = (
             db.session.query(tiers.variable_rate).
             join(details, tiers.chargeback_rate_detail_id == details.id).
             join(rates, details.chargeback_rate_id == rates.id).
             filter(details.metric == metric).
             filter(rates.rate_type == rate_type).
-            filter(rates.description == description)
+            filter(rates.description == description).all
         )
-    rate = db.session.query(tiers).filter(tiers.variable_rate.in_(
-        providers.subquery())).first().variable_rate
-    return rate
+    for row in result:
+        tiered_rate = {}
```

Comment:
Kind of a style comment, but consider this syntax here:

```python
tiered_rate = {
    'variable_rate': row.variable_rate,
    'fixed_rate': row.fixed_rate,
    'start': row.start,
    'finish': row.finish
}

or use a generator:

```python
{ var:getattr(row, var} for var in ['variable_rate', 'fixed_rate', 'start', 'finish'] }
```